### DISCUSSION TITLE
Use Rhino3D NURBS as input of examples

### DISCUSSION THREAD

**Discussion by Megidd** [mfem/mfem#4079] [Q&A]
I'm going to try to change the input of examples 1 and 2 and use some NURBS from [Rhino 7](https://www.rhino3d.com):

https://github.com/mfem/mfem/blob/4ba0c28a6ab815081946fdd7b5750be0324bc920/examples/ex1.cpp#L5

https://github.com/mfem/mfem/blob/4ba0c28a6ab815081946fdd7b5750be0324bc920/examples/ex2.cpp#L5

I have a problem domain I'm trying to explore.  For that, I'd prefer not going straight to code but instead exploring solutions using MFEM examples first. I'm totally blank 🙄 I was just wondering what's the standard way to take a NURBS from Rhino and use it as example inputs. I'm just lost and I'd appreciate any hint or help. Thanks.

**Comment by IdoAkkerman** [comment]
Great idea to make to allow NURBS from rhino as input.Please have a look at the nurbs miniapp. It has nurbs conversions of same of the examples.Verstuurd vanaf mijn iPhoneOp 23 jan. 2024 om 14:48 heeft Megidd ***@***.***> het volgende geschreven:﻿
I'm going to try to change the input of examples 1 and 2 and use some NURBS from Rhino 7:
https://github.com/mfem/mfem/blob/4ba0c28a6ab815081946fdd7b5750be0324bc920/examples/ex1.cpp#L5
https://github.com/mfem/mfem/blob/4ba0c28a6ab815081946fdd7b5750be0324bc920/examples/ex2.cpp#L5
I have a problem domain I'm trying to explore.  For that, I'd prefer not going straight to code but instead exploring solutions using MFEM examples first. I'm totally blank 🙄 I was just wondering what's the standard way to take a NURBS from Rhino and use it as example inputs. I'm just lost and I'd appreciate any hint or help. Thanks.

**Reply by Megidd** [reply]
> Great idea to make to allow NURBS from rhino as input. Please have a look at the nurbs miniapp. It has nurbs conversions of same of the examples.

Thanks @IdoAkkerman . I'm going to take a look at the NURBS mini-app and its conversions. I guess I may want to look here: https://github.com/mfem/mfem/tree/master/miniapps/nurbs

**Comment by IdoAkkerman** [comment]
@Megidd I do not know what your progress is on this project, but I just want to mention the following.
I know there are people doing IGA on Rhino geometries. But to my understanding they use (custom?) Rhino plugins to get the required data in the appropriate form.  This naturally leads to a fundamental choice whether you need to modify MFEM to read Rhino? Or write plugin for Rhino to output a MFEM mesh. Or a hybrid of these 2 extreme options.

I will try to ask around to see if people can share their plugins.

**Reply by Megidd** [reply]
Thanks @IdoAkkerman for the provided options.

I personally prefer to develop a BSD-licensed code for MFEM to import popular NURBS implementations like Rhino3D and more. If people can share their Rhino3D plugin code - probably in C# - then it can be a source of inspiration for developing the equivalent C++ code for MFEM and extending it.
